src/containers/App.js

```diff
@@ -50,27 +50,39 @@ const App = () => {
         }
     }));
 
-    const allStages = R.flatten(mockData.map((comp) => {
-        if(comp.stage !== undefined) {
-            return comp.stageName
-        }
+    const uniqueTags = R.uniq(allTags).filter(Boolean);
+
+    const stageMap = new Map();
+    R.flatten(mockData.map((comp) => {
+      if(comp.stage !== undefined){
+        return stageMap.set(comp.stageOrder, comp.stageName)
+      }
     }));
 
+    const uniqueStages = [];
 
-    const uniqueTags = R.uniq(allTags).filter(Boolean);
-    const uniqueStages = R.uniq(allStages).filter(Boolean);
+    mockData.forEach((comp) => {
+      uniqueStages[comp.stageOrder] = comp.stageName;
+    });
+
+    const lst = stageMap.keys()
+    const lst2 = Array.from(lst).sort();
+    console.log("number", lst2);
 
-     console.log('All unique Tags in the dataSet: ', allStages);
-    //console.log('and stages: ', uniqueStages);
+    const stagemapsorted = new Map();
+    for(let a = 0; a < lst2.length; a++){
+      stagemapsorted.set(lst2[a], stageMap.get(a+1))
+    }
```

Comment:
All on this, starting from 53st line looks too complicated. I'm pretty sure there should be more efficient way to do it, but i cant think of any right now :D